Ginger/GingerCoreNET/ActionsLib/ActUIElement.cs

```diff
@@ -906,9 +911,14 @@ public bool GetDisplayOutputValue()
             }
         }
 
-        public Drivers.CommunicationProtocol.PayLoad GetPayLoad()
+        public Drivers.CommunicationProtocol.PayLoad GetPayLoad(ActUIElement actUIElement)
         {
-            PayLoad PL = new PayLoad("UIElementAction");           
+            string payLoadName = @"UIElementAction";
+            if (Convert.ToBoolean(actUIElement.GetInputParamValue(Fields.IsWidgetsElement)))
```

Comment:
no need to pass act ui element. we are already in ActUIElement so you can do this.GetInputParamValue